### DISCUSSION TITLE
Max MB file size on phone

### DISCUSSION THREAD

**Discussion by gerritkk** [google/model-viewer#2716] [Q&A]
What is the max MB file size for the best AR experience on phone? For example I tried a 5mb .glb file on my iPhone 11 and it was very fast, but a 60mb .glb file took like 30 seconds. I prefer the loading time to be around 5 seconds or less.

**Comment by elalish** [comment]
There's no real answer to this; smaller is better! What seems reasonable on one network/device may be unreasonable on another. Know your users! And don't waste bytes. I find it's rarely necessary to use  more than a few mb to get high quality rendering on a mobile device as long as proper  compression  is used.

**Comment by rawwerks** [comment]
is anyone else seeing an error where large files just fail to launch AR completely? i have a ~35MB file that fails to launch AR, but smaller files are fine. the vertex count is ~400k - maybe it is just over the USDZ limit?

update: i fixed Quicklook by bringing vertex count to ~100k.

**Reply by rawwerks** [reply]
related: any tips on accessing the mesh vertex or face count inside model-viewer? this would help me create a prompt to warn the users that the file isn't likely going to go into AR.

or any recommendations for client-side JS that can compress meshes would be appreciated as well.

**Reply by elalish** [reply]
No, our APIs don't go that deep, but `gltf-validator` will tell you that. Also, it's generally the texture memory that really kills you; maybe just say any file >20mb is in the danger zone. It's likely device-dependent.

**Reply by vaggarath** [reply]
> 

Erm. I'm actually working with models over 150 Mo :D (i don't have a choice. It's reconstruction of CT scan) Well, loading is freaking long on low connections, but model-viewer works more than fine with those pachydermic models (at least on Android) ^^
